src/core/server/utils/workspace.ts

```diff
@@ -8,6 +8,7 @@ import { OpenSearchDashboardsRequest, ensureRawRequest } from '../http/router';
 export interface WorkspaceState {
   requestWorkspaceId?: string;
   isDashboardAdmin?: boolean;
+  isDataSourceAdmin?: boolean;
```

Comment:
I'd like to add a `TODO` here to move `isDataSourceAdmin` out of workspace state. We can do that in 2.18.

Reply:
Sure, I will add the todo comment